src/SigasiProjectCreator/ConverterHelper.py

```diff
@@ -4,149 +4,457 @@
+#     cygwin_process = subprocess.Popen(['/usr/bin/cygpath', '--windows', cygwin_path], stdout=subprocess.PIPE)
+#     cygwin_location = cygwin_process.communicate()[0].rstrip()
+#     return posixpath(cygwin_location)
 
 
-def convert_cygwin_path(cygwin_path):
-    cygwin_process = subprocess.Popen(['/usr/bin/cygpath', '--windows', cygwin_path], stdout=subprocess.PIPE)
-    cygwin_location = cygwin_process.communicate()[0].rstrip()
-    return posixpath(cygwin_location)
+def get_rel_or_abs_path(my_path: pathlib.Path, destination: pathlib.Path):
+    # If a relative path is given at this point, it is returned unchanged.
+    if not my_path.is_absolute():
+        return my_path
+    # If an absolute path is given and a relative path is expected, the relative path is returned.
+    destination_path = pathlib.Path(destination).absolute()
+    if my_path.is_relative_to(destination_path) or ArgsAndFileParser.get_use_relative_path(my_path):
+        # return input_path.relative_to(destination_path)
+        return Path(os.path.relpath(my_path, destination))
+    return my_path.absolute()
 
 
-def parse_and_create_project(usage, parse_file):
-    parser = ArgsAndFileParser(usage)
-    (project_name, _, destination, parser_output) = parser.parse_args_and_file(parse_file)
+virtual_folders = [pathlib.Path('Common Libraries')]
+
+
+def reset_for_unit_testing():
+    # We've used a few global variables, assuming that one run of SigasiProjectCreator creates only one project
+    # We need to reset those between unit tests
+    # TODO: implement a cleaner solution
+    global virtual_folders
+    virtual_folders = [pathlib.Path('Common Libraries')]
+    global linked_paths_simulator_layout
+    linked_paths_simulator_layout = []
+
+
+def check_and_create_virtual_folder(project_creator, file_name: pathlib.Path):
+    if not isinstance(file_name, pathlib.Path):
+        raise TypeError
+    filepath = file_name.parent
+    if filepath not in virtual_folders:
+        new_folders = []
+        while filepath and (str(filepath) != '.') and (filepath not in virtual_folders):
+            new_folders.insert(0, filepath)
+            filepath = filepath.parent
+        for new_path in new_folders:
+            virtual_folders.append(new_path)
+            project_creator.add_link(new_path, None, True)
+
+
+def check_and_create_linked_folder(project_creator, folder_name: pathlib.Path, folder_path: pathlib.Path):
+    if not (isinstance(folder_name, pathlib.Path) and isinstance(folder_path, pathlib.Path)):
+        raise TypeError
+    # TODO future work: map some folders into Common Libraries
+    # if folder_name.startswith('dependencies'):
+    #     virtual_path_name = posixpath(os.path.join('Common Libraries', folder_name))
+    check_and_create_virtual_folder(project_creator, folder_name)
+    project_creator.add_link(folder_name, get_rel_or_abs_path(folder_path, project_root), True)
+
+
+def uniquify_project_path(path: pathlib.Path, existing_path_list):
```

Comment:
I didn't know this word, but looking for a definition gives me 'remove duplicates from a list', which is not what this is doing.

Reply:
renamed to `get_unique_name`